src/main/java/ch/fhnw/algd1/converters/utf8/UTF8Converter.java

```diff
@@ -1,15 +1,41 @@
 package ch.fhnw.algd1.converters.utf8;
 
+import java.util.ArrayList;
+
 /*
  * Created on 05.09.2014
  */
 /**
- * @author 
+ * @author
  */
 public class UTF8Converter {
 	public static byte[] codePointToUTF(int x) {
 		byte[] b = null;
+		int firstStep = 0x80;
+		int secondStep = 2047;
+		int thirdStep = 0x10000;
+		int lastStep = 0x200000;
```

Comment:
Kann man so machen. Sie könnten aber auch die entsprechenden Zahlen direkt in die Fallunterscheidungen schreiben, da man die ja sonst nirgends braucht (und das auch nichts ist, dass sich im Laufe der Zeit ändern  wird...).
Seltsam (und im übrigen tatsächlich auch falsch) ist `2047` in dieser Liste.
Übrigens: Wenn Sie diese Werte _expliziter_ darstellen wollen, könnten Sie auch schreiben, `1 << 7`, `1 << 11` usw.